contracts/tests/keystone_forwarder.spec.ts

```diff
@@ -133,6 +158,19 @@ describe("keystone_storage", function () {
       "proposed owner set"
     );
 
+    const eventPromise = new Promise((resolve, reject) => {
```

Comment:
Because you're needing to do this pattern a couple times I'd recommend you make it in a utility function. That way we can use it in other tests as well. 

the function should accept 1. an event listener 2. callback function for encapsulating the assertion logic